examples/MQ/pixelAlternative/src/PixelAltFindHits.cxx

```diff
@@ -215,10 +214,12 @@ void PixelAltFindHits::ExecMQ(TList* inputList, TList* outputList)
     //  << "," << outputList->GetName() << "), Event " << fTNofEvents; LOG(info) <<
     //  "********************************************** PixelAltFindHits::ExecMQ(), Event " << fTNofEvents; LOG(info) <<
     //  "h" << FairLogger::flush;
-    fDigis = (TClonesArray*)inputList->FindObject("PixelDigis");
+    fDigis = dynamic_cast<TClonesArray*>(inputList->FindObject("PixelDigis"));
```

Comment:
For the digis I would propose to check also if the contained data class is the expected one. Later on we do a static_cast to this type.

fDigis = dynamic_cast<TClonesArray*>(inputList->FindObject("PixelDigis"));
  if (fDigis) {
    // Check if the stored data class has the right type  
    TString objectName = fDigis->GetClass()->GetName();
    if (0 != objectName.CompareTo("PixelDigi")) {
      throw std::runtime_error("no PixelDigis");
    }
else {
   throw std::runtime_error("no PixelDigi container");
}

Reply:
I don't like the string based things. I tried with `InheritsFrom` (which [TCollection::AssertClass](https://root.cern/doc/v628/classTCollection.html#a54d17025bcc640ac5a4f0ffa64cc819a) uses internally).